src/util.rs

```diff
@@ -242,3 +243,38 @@ pub fn short_file_path<'a>(target: &'a str, current_dir: &str) -> &'a str {
         .and_then(|p| p.to_str())
         .unwrap_or(target)
 }
+
+pub fn markdown_to_kakoune_markup<S: AsRef<str>>(markdown: S) -> String {
```

Comment:
`&str` would indeed not change much here. `AsRef` in general, has the benefit that you can pass a lot of values as-is and let the compiler/`AsRef` do the conversion, rather than having to do various `&`/`.as_str()`/etc. on the calling side.
It's also explicitly read-only.

`AsRef<str>` has become my go-to when I don't need an owned `String` after reading this: <https://phaazon.net/blog/on-owning-borrowing-pub-interface>.
`AsRef<[u8]>` is another nice one for "any byte-stream", whether that's a `File`, `String`, `Vec<u8>`, `[u8; <num>]` or something else.

Reply:
sounds good - on a second thought it's obvious that the parameter is not consumed, because it *only* `implements as_ref()`